src/tokenserver/extractors.rs

```diff
@@ -102,13 +111,46 @@ impl FromRequest for TokenserverRequest {
                 hashed_fxa_uid,
                 hashed_device_id,
                 service_id,
+                duration: duration.unwrap_or(DEFAULT_TOKEN_DURATION),
             };
 
             Ok(tokenserver_request)
         })
     }
 }
 
+#[derive(Deserialize)]
+struct QueryParams {
+    pub duration: Option<String>,
+}
+
+struct TokenDuration(Option<u64>);
+
+impl FromRequest for TokenDuration {
```

Comment:
nit: I'd argue `TokenDuration` doesn't warrant a struct and its own extractor if its to only be extracted once in `TokenserverRequest` but keeping it isn't a big deal.